sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java

```diff
@@ -150,41 +160,63 @@ static PropertyDefinition create(Property annotation) {
   }
 
   public static Result validate(PropertyType type, @Nullable String value, List<String> options) {
-    if (StringUtils.isNotBlank(value)) {
-      if (type == PropertyType.BOOLEAN) {
-        return validateBoolean(value);
-      } else if (type == PropertyType.INTEGER) {
-        return validateInteger(value);
-      } else if (type == PropertyType.FLOAT) {
-        return validateFloat(value);
-      } else if (type == PropertyType.SINGLE_SELECT_LIST && !options.contains(value)) {
-        return Result.newError("notInOptions");
-      }
+    if (isBlank(value)) {
+      return Result.SUCCESS;
     }
-    return Result.SUCCESS;
+
+    EnumMap<PropertyType, Validation> validations = createValidations(options);
+    return validations.getOrDefault(type, aValue -> Result.SUCCESS).validate(value);
   }
 
-  private static Result validateBoolean(String value) {
-    if (!StringUtils.equalsIgnoreCase(value, "true") && !StringUtils.equalsIgnoreCase(value, "false")) {
-      return Result.newError("notBoolean");
-    }
-    return Result.SUCCESS;
+  private static EnumMap<PropertyType, Validation> createValidations(List<String> options) {
+    EnumMap<PropertyType, Validation> validations = new EnumMap<>(PropertyType.class);
+    validations.put(BOOLEAN, validateBoolean());
+    validations.put(INTEGER, validateInteger());
+    validations.put(LONG, validateInteger());
+    validations.put(FLOAT, validateFloat());
+    validations.put(REGULAR_EXPRESSION, validateRegexp());
+    validations.put(SINGLE_SELECT_LIST, aValue -> options.contains(aValue) ? Result.SUCCESS : Result.newError("notInOptions"));
+    return validations;
   }
 
-  private static Result validateInteger(String value) {
-    if (!NumberUtils.isDigits(value)) {
-      return Result.newError("notInteger");
-    }
-    return Result.SUCCESS;
+  private static Validation validateBoolean() {
+    return value -> {
+      if (!StringUtils.equalsIgnoreCase(value, "true") && !StringUtils.equalsIgnoreCase(value, "false")) {
+        return Result.newError("notBoolean");
+      }
+      return Result.SUCCESS;
+    };
   }
 
-  private static Result validateFloat(String value) {
-    try {
-      Double.parseDouble(value);
+  private static Validation validateInteger() {
+    return value -> {
+      if (!NumberUtils.isDigits(value)) {
+        return Result.newError("notInteger");
+      }
       return Result.SUCCESS;
-    } catch (NumberFormatException e) {
-      return Result.newError("notFloat");
-    }
+    };
+  }
+
+  private static Validation validateFloat() {
+    return value -> {
+      try {
+        Double.parseDouble(value);
+        return Result.SUCCESS;
+      } catch (NumberFormatException e) {
+        return Result.newError("notFloat");
+      }
+    };
+  }
+
+  private static Validation validateRegexp() {
+    return value -> {
+      try {
+        Pattern.compile(value);
+        return Result.SUCCESS;
+      } catch (PatternSyntaxException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)